platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/Migratable.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+public interface Migratable {
+
+    /**
+    * Export all items to a directory
+    *
+    * @param exportPath - the path to the export directory
+    *
+    * @return - Collection of MigrationWarnings
+    */
+    public Collection<MigrationWarning> export(Path exportPath) throws MigrationException;
```

Comment:
Should we include a way to see progress for long running export operations?  Maybe events?